ui/dashboards/src/components/PanelDrawer/PanelEditorForm.tsx

```diff
@@ -65,10 +67,19 @@ export function PanelEditorForm(props: PanelEditorFormProps) {
     setGroupId(value);
   };
 
+  // TO DO: once we remove query editor from panel plugin, we can remove this method
+  const handlePluginSpecChange: OptionsEditorProps<UnknownSpec>['onChange'] = (spec) => {
+    // time series chart query editor returns spec.queries while gauge and stat charts return spec.query
+    if (spec?.queries || spec?.query) {
+      setQueries((spec?.queries ?? [spec?.query]) as QueryDefinition[]);
+    }
```

Comment:
Are there any edge cases you're missing where someone intentionally sets the queries to something falsy? (Not a leading question -- genuinely curious because I can't remember all the logistics of this code).